sessionctx/variable/session.go

```diff
@@ -819,7 +819,10 @@ func (s *SessionVars) CheckAndGetTxnScope() string {
 	if s.InRestrictedSQL {
 		return oracle.GlobalTxnScope
 	}
-	return s.TxnScope
+	if s.TxnScope == oracle.LocalTxnScope {
```

Comment:
I'm wondering, what if the zone value is happened to be `local` or `global`?

Reply:
I think it doesn't matter. Tidb will try find the local tso for `local` or `global` zone.

Reply:
We can return another boolean indicating whether it is local or global, leaving the returned string only valid if it is local scope.

Reply:
good catch, updated.